_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,652 @@
+{: .note }
+
+#### Sample request
+
+The following request creates a k-NN query that only returns the top hotels rated between 8 and 10, and that provide parking. The filter criteria is indicated with the Query DSL `range` query clause to indicate the range for the feedback ratings, and a `term` query clause to indicate "parking."
+
+```json
+POST /hotels-index/_search
+{
+    "size": 3,
+    "query": {
+        "knn": {
+            "location": {
+                "vector": [
+                    5.0,
+                    4.0
+                ],
+                "k": 3,
+                "filter": {
+                    "bool": {
+                        "must": [
+                            {
+                                "range": {
+                                    "rating": {
+                                        "gte": 8,
+                                        "lte": 10
+                                    }
+                                }
+                            },
+                            {
+                                "term": {
+                                    "parking": "true"
+                                }
+                            }
+                        ]
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+
+#### Sample Response
+
+The following response indicates that only three hotels met the filter criteria:
+
+
+```json
+{
+  "took" : 47,
+  "timed_out" : false,
+  "_shards" : {
+    "total" : 1,
+    "successful" : 1,
+    "skipped" : 0,
+    "failed" : 0
+  },
+  "hits" : {
+    "total" : {
+      "value" : 3,
+      "relation" : "eq"
+    },
+    "max_score" : 0.72992706,
+    "hits" : [
+      {
+        "_index" : "hotels-index",
+        "_id" : "3",
+        "_score" : 0.72992706,
+        "_source" : {
+          "location" : [
+            4.9,
+            3.4
+          ],
+          "parking" : "true",
+          "rating" : 9
+        }
+      },
+      {
+        "_index" : "hotels-index",
+        "_id" : "6",
+        "_score" : 0.3012048,
+        "_source" : {
+          "location" : [
+            6.4,
+            3.4
+          ],
+          "parking" : "true",
+          "rating" : 9
+        }
+      },
+      {
+        "_index" : "hotels-index",
+        "_id" : "5",
+        "_score" : 0.24154587,
+        "_source" : {
+          "location" : [
+            3.3,
+            4.5
+          ],
+          "parking" : "true",
+          "rating" : 8
+        }
+      }
+    ]
+  }
+}
+
+```
+
+## Additional complex filter query
+
+Depending on how restrictive you want your filter to operate, you can add multiple query types to a single request, such as: `term`, `wildcard`,  `regexp`, and `range`. You can then filter out the search results with the Boolean clauses `must`, `should`, and `must_not`.
```

Comment:
"Depending on how restrictive you want your filter, you can add multiple query types..."